src/main/java/net/objecthunter/exp4j/tokenizer/Tokenizer.java

```diff
@@ -133,6 +148,16 @@ private boolean isCloseParentheses(char ch) {
         return ch == ')' || ch == '}' || ch == ']';
     }
 
+    private Collection<String> undefinedVariables = null;
+    
+    public void allowUndefinedVariables(boolean allow) {
+    	this.undefinedVariables = (allow ? new ArrayList<String>() : null);
+    }
+    
+    public String[] getUndefinedVariables() {
+    	return (undefinedVariables != null ? undefinedVariables.toArray(new String[undefinedVariables.size()]) : null);
```

Comment:
This I didn't know, I always assumed allocating the right size was best! Thanks for the "wisdom"!